releasePipeline/33-build-resources-image.sh

```diff
@@ -0,0 +1,166 @@
+}
+info() { printf "${white}➜ %s${reset}\n" "$@"
+}
+success() { printf "${green}✔ %s${reset}\n" "$@"
+}
+error() { printf "${red}✖ %s${reset}\n" "$@"
+}
+warn() { printf "${tan}➜ %s${reset}\n" "$@"
+}
+bold() { printf "${bold}%s${reset}\n" "$@"
+}
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@"
+}
+
+
+#-----------------------------------------------------------------------------------------                   
+# Main logic.
+#-----------------------------------------------------------------------------------------   
+
+mkdir -p ${WORKSPACE_DIR}/temp
+
+
+function get_galasa_version_to_be_released {
+    h1 "Working out the version of Galasa to test and release."
+
+    url="https://development.galasa.dev/main/maven-repo/obr/dev/galasa/dev.galasa.uber.obr/"
+    curl $url > temp/galasa-version.txt -s
+    rc=$?; 
+    if [[ "${rc}" != "0" ]]; then 
+      error "Failed to get galasa version"
+      exit 1
+    fi
+
+    # Note: We take the 2nd line which has an "<a href" string on... hopefully it won't change...
+    galasa_version=$(cat temp/galasa-version.txt | grep "<a href" | head -2 | tail -1 | cut -f2 -d'"' | cut -f1 -d'/')
+
+    success "Galasa version to be tested and released is ${galasa_version}"
+    export galasa_version
+}
+
+
+function set_kubernetes_context {
+    h1 "Setting the kubernetes context to be cicsk8s, using namespace galasa-build"
+    kubectl config set-context cicsk8s --namespace=galasa-build
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed. rc=${rc}"
+        exit 1
+    fi
+    
+}
+
+function create_resources_image {
+
+    h1 "Creating the resources image..."
+
+    cd ${WORKSPACE_DIR}/temp
+    yaml_file=${WORKSPACE_DIR}/temp/build-resources-images.yaml
+    cat << EOF > $yaml_file
+#
+# Copyright contributors to the Galasa project 
+#
+kind: PipelineRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+  generateName: resources-
+  annotations:
+    argocd.argoproj.io/compare-options: IgnoreExtraneous
+    argocd.argoproj.io/sync-options: Prune=false
+#
+#
+#
+spec:
+#
+#
+#
+  pipelineRef:
+    name: resources-build
+
+  serviceAccountName: galasa-build-bot
+
+  params:
+  - name: distBranch
+    value: "release"
+  - name: version
+    value: "${galasa_version}"
+#
+#
+#
+  workspaces:
+  - name: git-workspace
+    volumeClaimTemplate:
+      spec:
+        storageClassName: longhorn-temp
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 20Gi
+EOF
+
+    cmd="kubectl -n galasa-build create -f $yaml_file"
+    info "Command is $cmd"
+    $cmd
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then
+        error "Failed to create the resources image. rc=$rc"
+        exit 1
+    fi
+
+    success "Resources image build pipeline kicked off OK."
+}
+
+set_kubernetes_context
+get_galasa_version_to_be_released
+
+create_resources_image
+
+note "Now wait for the 'resources-*' pipeline to complete."
+
```

Comment:
We can check the status of the pipelineRun similar to how we did in 03-repo-braches-delete.sh and 04-repo-branches-create.sh?

Reply:
Don't know. Improvement for later ? Raise a story ? I think we should be parameterising the pipeline and using `tkn` to kick things off instead of `kubectl apply -f` anyway, which appears to have a 'wait' kind of option.

Reply:
Not going to do that in this PR as I need to get it merged and we can go from there.